google/cloud/storage/internal/async/writer_connection_impl.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/async/writer_connection_impl.h"
+#include "google/cloud/storage/internal/async/partial_upload.h"
+#include "google/cloud/storage/internal/async/write_payload_impl.h"
+#include "google/cloud/storage/internal/grpc/ctype_cord_workaround.h"
+#include "google/cloud/storage/internal/grpc/object_metadata_parser.h"
+#include "google/cloud/internal/make_status.h"
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+auto HandleFinishAfterError(std::string msg) {
+  return [m = std::move(msg)](future<Status> f) {
+    auto status = f.get();
+    if (!status.ok()) return status;
+    // A `Write()` or `Read()` operation unexpectedly returned `ok == false`.
+    // The `Finish()` call should return the error detailed, it returned
+    // "success". This is some kind of internal error in the client library,
```

Comment:
Perhaps _"... should return the detailed error, but it returned "success"."_?

Reply:
Fixed.